### ISSUE TITLE
Segmentation fault in multi-objective constrainedbased

### ISSUE THREAD

**Issue by Chickenpowerrr** [moves-rwth/storm#625]
When evaluating the property `multi(P>=0 [F "valueCollected"]))`  on the [Mars rover model](https://github.com/tquatmann/qcomp23-multi/blob/v1.0/models/rov/rov.prism), using the constants `B=10,Unf=2` with the constraintbased solver, Storm exists with a segmentation fault.

```console
Storm 1.9.0

Date: Sat Oct 19 15:31:01 2024
Command line arguments: --prism rov.prism -prop 'multi(P>=0 [F "valueCollected"])' '--multiobjective:method' constraintbased -const 'B=10,Unf=2'
Current working directory:

Time for model input parsing: 0.005s.

Time for model construction: 0.364s.

--------------------------------------------------------------
Model type:     MDP (sparse)
States:         161410
Transitions:    302642
Choices:        201762
Reward Models:  none
State Labels:   3 labels
   * deadlock -> 0 item(s)
   * init -> 1 item(s)
   * valueCollected -> 6770 item(s)
Choice Labels:  none
--------------------------------------------------------------

Model checking property "1": multi(P>=0 [F "valueCollected"]) ...
Segmentation fault (core dumped)
```

**Comment by sjunges** [comment]
Hey,

thanks for the error message, that definitively should not happen! 
Does Storm also crash for any nontrivial property (i.e., for e.g. Prob >= 0.0001) ? 

Best,
Sebastian

**Comment by Chickenpowerrr** [comment]
Hi Sebastian,

Yes, the bound does not seem to influence the segmentation fault.

Kind regards,

Mark

**Comment by tquatmann** [comment]
This is  a  stackoverflow happening when translating from Storm's expressions to Z3 expressions. [Z3](https://github.com/Z3Prover/z3) is the SMTSolver invoked here, which does the actual hard work of deciding if the constraint system is satisfiable.

I just pushed a fix in PR #626.

Notice that the constraint-based approach does not scale well to larger models. While my fix avoids a stackoverflow, you likely get a timeout now.
